File: sdks/api/src/main/java/com/vivid/clients/api/metadata/StringListMetadataValue.java
package com.vivid.clients.api.metadata;

import com.vivid.clients.api.MetadataValue;

import java.util.List;

public record StringListMetadataValue(List<String> content) implements MetadataValue<List<String>> {
    @Override
    public List<String> getContent() {
        return content;
    }
}

File: sdks/kotlin/src/main/kotlin/com/vivid/sdk/api/metadata/StringListMetadataValue.kt
package com.vivid.sdk.api.metadata

import com.vivid.sdk.api.MetadataValue

/**
 * Metadata value containing a list of strings.
 *
 * @property content the list of strings
 */
data class StringListMetadataValue(
    val content: List<String>
) : MetadataValue {
    fun contains(value: String): Boolean {
        return content.contains(value)
    }
}
